src/datasets.jl

```diff
@@ -312,29 +312,50 @@ function datasets(
     auth::Authentication=__auth__(),
 )
     datasets, _ = try
-        _get_datasets(; auth)
+        _get_datasets(auth)
     catch e
         e isa JuliaHubException && rethrow(e)
         e isa JuliaHubError && rethrow(e)
         throw(
             JuliaHubError("Error while retrieving datasets from the server", e, catch_backtrace()),
         )
     end
-    shared || filter!(datasets) do dataset
-        dataset["owner"]["username"] == username
+    # It might happen that some of the elements of the `datasets` array can not be parsed for some reason,
+    # and the Dataset() constructor will throw. Rather than having `datasets` throw an error (as we would
+    # normally do for invalid backend responses), in this case we handle the situation more gracefully,
+    # and just filter out the invalid elements (and print a warning to the user -- it's still an unusual
+    # situation and should be reported). This is because we also re-use `JuliaHub.datasets` in
+    # `JuliaHub.dataset` when fetching the information for just one dataset, and unconditionally throwing
+    # would mean that `JuliaHub.dataset` can break due an issue with an unrelated dataset.
+    n_erroneous_datasets = 0
+    datasets = map(datasets) do dataset
+        try
+            # We also use the `nothing` method for filtering out datasets that are not owned by the
+            # current `username` if `shared = false`.
+            if !shared && (dataset["owner"]["username"] != username)
+                return nothing
+            end
+            return Dataset(dataset)
+        catch e
+            @debug "Invalid dataset in GET /datasets response" dataset exception = (
+                e, catch_backtrace()
+            )
+            n_erroneous_datasets += 1
+            return nothing
+        end
     end
-    return try
-        Dataset.(datasets)
-    catch e
-        throw(JuliaHubError("Unable to parse the response from the server", e, catch_backtrace()))
+    if n_erroneous_datasets > 0
+        @warn "The JuliaHub GET /datasets response had $(n_erroneous_datasets) erroneous dataset(s)."
```

Comment:
```suggestion
        @warn "The JuliaHub GET /datasets response contains erroneous dataset(s). Omitting $(n_erroneous_datasets) entries."
```
to make it a bit clearer that these are not shown?